crates/ruff/src/rules/pylint/rules/continue_in_finally.rs

```diff
@@ -0,0 +1,38 @@
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::define_violation;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+use rustpython_parser::lexer;
+use rustpython_parser::lexer::Tok;
+
+use ruff_macros::derive_message_formats;
+use rustpython_ast::Stmt;
+
+define_violation!(
+    pub struct ContinueInFinally;
+);
+impl Violation for ContinueInFinally {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Continue is not supported inside a finally block")
+    }
+}
+
+/// PLE0116
+pub fn continue_in_finally(checker: &mut Checker, finalbody: &[Stmt]) {
+    if finalbody.is_empty() {
+        return;
+    }
+    let first = finalbody.first().unwrap();
+    let last = finalbody.last().unwrap();
+    let range = Range::new(first.location, last.end_location.unwrap());
+    let contents = checker.locator.slice_source_code_range(&range);
+    for (_, tok, _) in lexer::make_tokenizer(contents).flatten() {
```

Comment:
I think the way to get the right location would be to use `lexer::make_tokenizer_located(contents, range.location)`, and then use the `start` and `end` of `(start, tok, end`).

Reply:
That being said: I think we should instead do this with a visitor and the AST, rather than re-tokenzing. Take a look at how `ReturnStatementVisitor` works. We should try to find a `StmtKind::Continue` in `finalbody`.

Reply:
I got this solution implemented!